File: HighSpiritApp/Models/Locker/Locker.cs
using System.ComponentModel.DataAnnotations;

namespace HighSpiritApp.Models.Locker
{
    /// <summary>
    /// Locker entity for gym locker management
    /// </summary>
    public class Locker
    {
        [Key]
        public int LockerID { get; set; }

        [Required]
        [StringLength(20)]
        public string LockerNumber { get; set; } = null!;

        [Required]
        public string Gender { get; set; } = "Gents"; // Gents, Ladies

        public string Status { get; set; } = "Empty"; // Empty, Occupied, Locked, Maintenance

        // Assigned member info (nullable when not assigned)
        public int? CustomerID { get; set; }
        public string? AssignedTo { get; set; } // Member name
        public string? AssignedPhone { get; set; } // Member phone

        // Package/Plan linked - stores the full package name from customer
        public string? Package { get; set; } // e.g., "Custom 2 - Gym & Cardio", "Gym Package", etc.

        // Rental details
        public DateTime? StartDate { get; set; }  // Joined Date
        public DateTime? EndDate { get; set; }    // Expiry
        public int RentalMonths { get; set; }     // Duration in months
        public decimal MonthlyRate { get; set; }
        public decimal TotalAmount { get; set; }  // Amount
        public decimal PaidAmount { get; set; }
        public decimal DueAmount { get; set; }    // Due Amount

        public string? Remarks { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime? UpdatedAt { get; set; }

        // Computed properties
        public bool IsExpired => EndDate.HasValue && EndDate.Value < DateTime.Today;
        public bool IsExpiringSoon => EndDate.HasValue && EndDate.Value >= DateTime.Today && EndDate.Value <= DateTime.Today.AddDays(7);
        public int DaysRemaining => EndDate.HasValue ? Math.Max(0, (EndDate.Value - DateTime.Today).Days) : 0;
    }
}

File: high_spirit_gym/lib/models/locker.dart
class GymLocker {
  final int lockerID;
  final String lockerNumber;
  final String gender;
  final String status;
  final int? customerID;
  final String? assignedTo;
  final String? assignedPhone;
  final String? package;
  final DateTime? startDate;
  final DateTime? endDate;
  final int rentalMonths;
  final double monthlyRate;
  final double totalAmount;
  final double paidAmount;
  final double dueAmount;
  final String? remarks;
  final bool isExpired;
  final bool isExpiringSoon;
  final int daysRemaining;

  GymLocker({
    required this.lockerID,
    required this.lockerNumber,
    required this.gender,
    required this.status,
    this.customerID,
    this.assignedTo,
    this.assignedPhone,
    this.package,
    this.startDate,
    this.endDate,
    required this.rentalMonths,
    required this.monthlyRate,
    required this.totalAmount,
    required this.paidAmount,
    required this.dueAmount,
    this.remarks,
    required this.isExpired,
    required this.isExpiringSoon,
    required this.daysRemaining,
  });

  factory GymLocker.fromJson(Map<String, dynamic> json) {
    return GymLocker(
      lockerID: json['lockerID'] ?? 0,
      lockerNumber: json['lockerNumber'] ?? '',
      gender: json['gender'] ?? 'Gents',
      status: json['status'] ?? 'Empty',
      customerID: json['customerID'],
      assignedTo: json['assignedTo'],
      assignedPhone: json['assignedPhone'],
      package: json['package'],
      startDate: json['startDate'] != null ? DateTime.tryParse(json['startDate']) : null,
      endDate: json['endDate'] != null ? DateTime.tryParse(json['endDate']) : null,
      rentalMonths: json['rentalMonths'] ?? 0,
      monthlyRate: (json['monthlyRate'] as num?)?.toDouble() ?? 0,
      totalAmount: (json['totalAmount'] as num?)?.toDouble() ?? 0,
      paidAmount: (json['paidAmount'] as num?)?.toDouble() ?? 0,
      dueAmount: (json['dueAmount'] as num?)?.toDouble() ?? 0,
      remarks: json['remarks'],
      isExpired: json['isExpired'] ?? false,
      isExpiringSoon: json['isExpiringSoon'] ?? false,
      daysRemaining: json['daysRemaining'] ?? 0,
    );
  }
}
